pkg/collector/collector.go

```diff
@@ -45,7 +45,43 @@ type Collector struct {
 	databaseAge *prometheus.Desc
 }
 
-//New creates a Collector struct
+type VersionInfo struct {
+	clamav_version   string
+	database_version string
+	database_age     float64
+	versions_parsed  bool
+}
+
+func GetVersionInfo(version_string string) *VersionInfo {
+	versionInfo := VersionInfo{}
+	// remove newlines
+	version := strings.Replace(version_string, "\n", "", -1)
+	regex := regexp.MustCompile(`((ClamAV)+\s([0-9.]*)\/?([0-9.]*)?\/?(\w{1,4} \w{1,4} \d{1,2} \d{1,2}:\d{1,2}:\d{1,2} \d{4})?)`)
+	matches := regex.FindAllStringSubmatch(string(version), -1)
+	strBuilddate := ""
+
+	// Parse version numbers
+	if len(matches) > 0 {
+		versionInfo.clamav_version = matches[0][3]
+		versionInfo.database_version = matches[0][4]
+		strBuilddate = matches[0][5]
+		versionInfo.versions_parsed = true
+	}
+
+	// Parse string as date type
+	dateFmt := "Mon Jan 2 15:04:05 2006"
+	builddate, err := time.Parse(dateFmt, strBuilddate)
+
+	if err != nil {
+		versionInfo.database_age = float64(time.Now().Unix())
```

Comment:
It feels wrong to set this to now, but I understand that's setting it to max age. Can you set it to `math.NaN()` instead?

Reply:
It probably makes sense to log using `log.Error("Error parsing ClamAV date: ", err)` in this case even though its weird to log in a helper function but I'm not sure where else you would